### PR TITLE
binding: add BindPlain function

### PR DESCRIPTION
**Pull request by guonaihong**
```go
package main

import (
	"github.com/gin-gonic/gin"
)

func main() {
	router := gin.Default()
	router.POST("/", func(c *gin.Context) {
		var s string
		c.BindPlain(&s)
		c.String(200, s)
	})

	router.Run()
}

// client
// curl -d "test string" 127.0.0.1:8080
// output
// test string
```

### CODE REVIEW

**Comment by guonaihong** [comment]
Can the pr review it quickly?
If you trust the client to send a json string, you just want to forward it to the next server node. There is no need to use BindJSON functions, it is more efficient to use BindPlain.

@appleboy @thinkerou

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+*~
```
Comment (appleboy) [inline .gitignore]: revert this

**Comment by appleboy** [comment]
move to 1.7 milestone

File: .gitignore (hunk 1 above)
Comment (guonaihong) [reply .gitignore]: ok

**Comment by appleboy** [comment]
Please fix the conflicts and move to 1.11

**Comment by guonaihong** [comment]
Okay, I'll set aside time on Saturday to fix it.

**Comment by guonaihong** [comment]
I might only have time to get this done on Monday evening

**Comment by appleboy** [comment]
Please help to rebase the latest master branch. Thanks.

**Comment by guonaihong** [comment]
> Please help to rebase the latest master branch. Thanks.

ok

**Comment by appleboy** [comment]
@guonaihong Can you close this PR and create a new PR? Since there is no CI/CD (testing and lint) checking in this PR.

**Comment by guonaihong** [comment]
ok, I'll close this PR for now and open a new one on Monday.

**Pull request closed without merging**